storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -166,93 +257,182 @@ func (s *commitLogSource) Read(
 					lastWriteAt: dp.Timestamp,
 					enc:         enc,
 				})
-				unmergedSeries.encoders[blockStart] = unmergedBlock
+				unmergedSeries.encoders[blockStart.UnixNano()] = unmergedBlock
 			}
 		}
 		if err != nil {
-			errs++
+			workerErrs[workerNum]++
 		}
 	}
-	if errs > 0 {
-		s.log.Errorf("error bootstrapping from commit log: %d block encode errors", errs)
+	wg.Done()
+}
+
+func (s *commitLogSource) shouldEncodeSeries(
+	namespace ts.ID,
+	shardsTimeRanges result.ShardTimeRanges,
+	blockSize time.Duration,
+	series commitlog.Series,
+	dp ts.Datapoint,
+) bool {
+	// Check if the series belongs to current namespace being bootstrapped
+	if !namespace.Equal(series.Namespace) {
+		return false
+	}
+
+	// Check if the shard is one of the shards we're trying to bootstrap
+	ranges, ok := shardsTimeRanges[series.Shard]
+	if !ok {
+		return false
+	}
+
+	// Check if the block corresponds to the time-range that we're trying to bootstrap
+	blockStart := dp.Timestamp.Truncate(blockSize)
+	blockEnd := blockStart.Add(blockSize)
+	blockRange := xtime.Range{
+		Start: blockStart,
+		End:   blockEnd,
+	}
+
+	return ranges.Overlaps(blockRange)
+}
+
+func (s *commitLogSource) printM3TSZEncodingOutcome(workerErrs []int, iter commitlog.Iterator) {
```

Comment:
Can we change print... to log... to be more clear (i.e. we're not just printing to stdout/stderr, depending on where our log goes it goes to wherever it is directed to).  We could call this simply `logEncodingOutcome(..)` again because encoders could technically also be an encoder other than a M3TSZ encoder.